### ISSUE TITLE
Is it possible to remove errors with EMI?

### ISSUE THREAD

**Issue by Lolothepro** [Soaryn/XyCraftTracker#51]
### Mod Title

XyCraft Core

### Minecraft Version

1.21.0

### Severity

Error in log

### NeoForge Version

21.0.64-beta

### Describe the Suggestion

```
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] Exception loading plugin provided by xycraft_machines
java.lang.UnsupportedOperationException: Cannot mutate an empty stack
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.lang.UnsupportedOperationException: Cannot mutate an empty stack
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] emi@1.1.10+1.21+neoforge
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] dev.emi.emi.api.stack.EmptyEmiStack.setRemainder(EmptyEmiStack.java:34)
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] xycraft_machines@0.7.28
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] tv.soaryn.xycraft.machines.compat.viewers.emi.EMIExtractorRecipe.lambda$getEmiIngredientStream$0(EMIExtractorRecipe.java:58)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:174)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] xycraft_machines@0.7.28
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] tv.soaryn.xycraft.machines.compat.viewers.emi.EMIExtractorRecipe.<init>(EMIExtractorRecipe.java:41)
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] xycraft_machines@0.7.28
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] tv.soaryn.xycraft.machines.compat.viewers.emi.EMIPlugin.lambda$register$0(EMIPlugin.java:28)
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] xycraft_machines@0.7.28
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] tv.soaryn.xycraft.machines.compat.viewers.ExtractorViewerCommon.handleIntegration(ExtractorViewerCommon.java:102)
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] xycraft_machines@0.7.28
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] tv.soaryn.xycraft.machines.compat.viewers.emi.EMIPlugin.register(EMIPlugin.java:22)
	at TRANSFORMER
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] emi@1.1.10+1.21+neoforge
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] dev.emi.emi.runtime.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:185)
	at java.base
[15:56:03] [Thread-30/ERROR] [EMI/]: [EMI] java.lang.Thread.run(Thread.java:1583)

[15:56:03] [Thread-30/INFO] [EMI/]: [EMI] Loading plugin from ae2
[15:56:03] [Server thread/WARN] [minecraft/MinecraftServer]: Can't keep up! Is the server overloaded? Running 3353ms or 67 ticks behind
	at TRANSFORMER/emi@1.1.10+1.21+neoforge/dev.emi.emi.api.stack.EmptyEmiStack.setRemainder(EmptyEmiStack.java:34)
	at TRANSFORMER/xycraft_machines@0.7.28/tv.soaryn.xycraft.machines.compat.viewers.emi.EMIExtractorRecipe.lambda$getEmiIngredientStream$0(EMIExtractorRecipe.java:58)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:174)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at TRANSFORMER/xycraft_machines@0.7.28/tv.soaryn.xycraft.machines.compat.viewers.emi.EMIExtractorRecipe.<init>(EMIExtractorRecipe.java:41)
	at TRANSFORMER/xycraft_machines@0.7.28/tv.soaryn.xycraft.machines.compat.viewers.emi.EMIPlugin.lambda$register$0(EMIPlugin.java:28)
	at TRANSFORMER/xycraft_machines@0.7.28/tv.soaryn.xycraft.machines.compat.viewers.ExtractorViewerCommon.handleIntegration(ExtractorViewerCommon.java:102)
	at TRANSFORMER/xycraft_machines@0.7.28/tv.soaryn.xycraft.machines.compat.viewers.emi.EMIPlugin.register(EMIPlugin.java:22)
	at TRANSFORMER/emi@1.1.10+1.21+neoforge/dev.emi.emi.runtime.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:185)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

This error with EMI

### Reproduction Steps

Launch EMI with XyCraft Core

**Comment by Soaryn** [comment]
I'll look into it, I've cleaned up quite a bit, but there is still quite a bit of work to do. Errors though, I wasn't getting on the version I was on (at least in dev now)

**Comment by Soaryn** [comment]
I'm not seeing the errors on my side on load, but I've added an `isEmpty` check just for good measure :) Will be resolved in `0.7.29`
